### PR TITLE
docs: Fixed Typos in GSoC Guide

### PR DESCRIPTION
**Pull request by kaustubhgupta**
Fixes: #1084 

This PR fixes some of the minor typos in the GSoC guide.

### CODE REVIEW

File: docs/contributing/gsoc/2021/index.rst [hunk 1]
```diff
@@ -9,12 +9,12 @@ https://summerofcode.withgoogle.com/how-it-works/#timeline
 About DFFML
 -----------
 
-DFFML is a plugin based library / framework for machine learning. It allows
+DFFML is a plugin-based library/framework for machine learning. It allows
 users to wrap high or low level implementations of models that use various
-machine learning libraries, so as to interact will lots of different model
+machine learning libraries, so as to interact will lot of different model
```
Comment (programmer290399) [inline docs/contributing/gsoc/2021/index.rst:14]: I think `lots` should remain as it is and `will` in this sentence should become `with`

File: docs/contributing/gsoc/2021/index.rst [hunk 2]
```diff
@@ -9,12 +9,12 @@ https://summerofcode.withgoogle.com/how-it-works/#timeline
...
 implementations in the same way.
 
-DFFML is also a tool for dataset generation. DFFML ues directed graphs to
+DFFML is also a tool for dataset generation. DFFML use directed graphs to
```
Comment (programmer290399) [inline docs/contributing/gsoc/2021/index.rst:17]: Shouldn't it be `uses` instead of `use` ?

File: docs/contributing/gsoc/2021/index.rst [hunk 3]
```diff
@@ -115,7 +110,7 @@ Deadlines
 See the milestones prefixed with "GSoC 2021"
 https://github.com/intel/dffml/milestones?direction=desc&sort=due_date&state=open
 
-Please see the :doc:`/contributing/gsoc/rubric` page for the project
+Please see the: doc:`/contributing/gsoc/rubric` page for the project
```
Comment (programmer290399) [inline docs/contributing/gsoc/2021/index.rst:113]: `:doc:` actually makes it a link to the path after it in the docs, we don't need to move the ":" ,  otherwise it would link to the desired page , you'd need to revert this change :)

File: docs/contributing/gsoc/2021/index.rst (hunk 1 above)
Comment (kaustubhgupta) [reply docs/contributing/gsoc/2021/index.rst:14]: Yes, that sounds better

File: docs/contributing/gsoc/2021/index.rst (hunk 2 above)
Comment (kaustubhgupta) [reply docs/contributing/gsoc/2021/index.rst:17]: okay

File: docs/contributing/gsoc/2021/index.rst (hunk 3 above)
Comment (kaustubhgupta) [reply docs/contributing/gsoc/2021/index.rst:113]: okay got it

File: docs/contributing/gsoc/2021/index.rst (hunk 1 above)
Comment (kaustubhgupta) [reply docs/contributing/gsoc/2021/index.rst:14]: Yes, that sounds better

File: docs/contributing/gsoc/2021/index.rst (hunk 2 above)
Comment (kaustubhgupta) [reply docs/contributing/gsoc/2021/index.rst:17]: okay will do that

File: docs/contributing/gsoc/2021/index.rst (hunk 3 above)
Comment (kaustubhgupta) [reply docs/contributing/gsoc/2021/index.rst:113]: oh, okay got it

**Comment by kaustubhgupta** [comment]
@programmer290399 I have made the changes as suggested

**Comment by kaustubhgupta** [comment]
@programmer290399?
